internal/util/helpers_test.go

```diff
@@ -4,18 +4,767 @@
+				t.Logf("Test %s: written file path %s with content: %s", tt.name, filePath, tt.fileContent)
+			} else {
+				// Print the file path for debugging
+				t.Logf("Test %s: file path %s not created as file content is empty", tt.name, filePath)
+			}
+
+			// Temporarily override the environment variable for the test
+			originalEnv := os.Getenv("XDG_CONFIG_HOME")
+			err = os.Setenv("XDG_CONFIG_HOME", tempDir)
+			if err != nil {
+				t.Errorf("error setting XDG_CONFIG_HOME: %v", err)
+			}
+
+			result, err := util.LoadCredentials()
+			if tt.expectedError != "" {
+				if err == nil || !strings.HasPrefix(err.Error(), tt.expectedError) {
+					t.Errorf("expected error matching %v, got %v", tt.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+				if result.AccessToken != tt.expectedResult.AccessToken || result.RefreshToken != tt.expectedResult.RefreshToken || !result.AccessTokenExpiresAt.Equal(tt.expectedResult.AccessTokenExpiresAt) {
+					t.Errorf("expected result %v, got %v", tt.expectedResult, result)
+				}
+			}
+			defer os.Setenv("XDG_CONFIG_HOME", originalEnv)
+			defer os.RemoveAll(tempDir)
+		})
+
+	}
+}
+
+// TestRevokeToken tests the RevokeToken function
+func TestRevokeToken(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name         string
+		token        string
+		issuerUrl    string
+		clientId     string
+		tokenHint    string
+		expectedPath string
+		expectError  bool
+	}{
+		{
+			name:         "Valid token revocation",
+			token:        "test-token",
+			issuerUrl:    "http://localhost:8081",
+			clientId:     "minder-cli",
+			tokenHint:    "refresh_token",
+			expectedPath: "/realms/stacklok/protocol/openid-connect/revoke",
+			expectError:  false,
+		},
+		{
+			name:         "Invalid issuer URL",
+			token:        "test-token",
+			issuerUrl:    "://invalid-url",
+			clientId:     "minder-cli",
+			tokenHint:    "refresh_token",
+			expectedPath: "",
+			expectError:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			var server *httptest.Server
+			if tt.name != "Invalid issuer URL" {
+				server = httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+					if r.URL.Path != tt.expectedPath {
+						t.Errorf("expected path %s, got %s", tt.expectedPath, r.URL.Path)
+					}
+
+					if err := r.ParseForm(); err != nil {
+						t.Fatalf("error parsing form: %v", err)
+					}
+
+					if r.Form.Get("client_id") != tt.clientId {
+						t.Errorf("expected client_id %s, got %s", tt.clientId, r.Form.Get("client_id"))
+					}
+
+					if r.Form.Get("token") != tt.token {
+						t.Errorf("expected token %s, got %s", tt.token, r.Form.Get("token"))
+					}
+
+					if r.Form.Get("token_type_hint") != tt.tokenHint {
+						t.Errorf("expected token_type_hint %s, got %s", tt.tokenHint, r.Form.Get("token_type_hint"))
+					}
+				}))
+				defer server.Close()
```

Comment:
You can't put this inside a constructor function if you take that approach.

Reply:
I addressed this as I have changed the approach with test server creation.